### PR TITLE
Merge Fixes From Dev-Branch

### PR DESCRIPTION
**Pull request by kapilash**
This PR brings in two fixes in the dev branch that were missing from the master - #6989 and #6991

Also included are pending PR comments from #7031

### CODE REVIEW

**Comment by kapilash** [comment]
cc @stephentoub @shrutigarg

**Comment by joshfree** [comment]
@ericeil can you take a look?

**Comment by ericeil** [comment]
Is it possible to submit this as three separate commits: one for each of the two merged commits, and one for the fixes for #7031?  It would make it easier to review now, and easier to track code flow later.

File: src/Native/System.Net.Security.Native/pal_gssapi.cpp [hunk 1]
```diff
@@ -35,9 +35,13 @@ static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
 
 #if !HAVE_GSS_SPNEGO_MECHANISM
 static char gss_spnego_oid_value[] = "\x2b\x06\x01\x05\x05\x02"; // Binary representation of SPNEGO Oid (RFC 4178)
-static gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .elements = static_cast<void*>(gss_spnego_oid_value)};
+static gss_OID_desc gss_mech_spnego_OID_desc = {
+    .length = static_cast<OM_uint32>(sizeof(gss_spnego_oid_value) / sizeof(gss_spnego_oid_value[0]) - 1),
```
Comment (ericeil) [inline src/Native/System.Net.Security.Native/pal_gssapi.cpp]: It looks like we have an `ARRAY_SIZE` macro that would make this more clear:

```
    .length = ARRAY_SIZE(gss_spnego_oid_value) - 1,
```

File: src/Native/System.Net.Security.Native/pal_gssapi.cpp [hunk 2]
```diff
@@ -35,9 +35,13 @@ static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
...
+    .elements = static_cast<void*>(gss_spnego_oid_value)};
 static char gss_ntlm_oid_value[] = "\x2b\x06\x01\x04\x01\x82\x37\x02\x02\x0a"; // Binary representation of NTLM OID (https://msdn.microsoft.com/en-us/library/cc236636.aspx)
-static gss_OID_desc gss_mech_ntlm_OID_desc = { .length = 10, .elements = static_cast<void*>(gss_ntlm_oid_value) };
+static gss_OID_desc gss_mech_ntlm_OID_desc = { 
+    .length = static_cast<OM_uint32>(sizeof(gss_ntlm_oid_value) / sizeof(gss_ntlm_oid_value[0]) - 1),
```
Comment (ericeil) [inline src/Native/System.Net.Security.Native/pal_gssapi.cpp]: Use `ARRAY_SIZE` here as well.

**Comment by ericeil** [comment]
Aside from a couple of nits, this LGTM.  @stephentoub or @bartonjs might have more context, and should also have a look.

**Comment by kapilash** [comment]
> Is it possible to submit this as three separate commits: one for each of the two merged commits, and one for the fixes for #7031? It would make it easier to review now, and easier to track code flow later.

Sure. Rebased it with master and converted it to three commits.

File: src/Native/System.Net.Security.Native/pal_gssapi.cpp (hunk 1 above)
Comment (kapilash) [inline src/Native/System.Net.Security.Native/pal_gssapi.cpp]: done. Thank you. Did not know of the macro

**Comment by stephentoub** [comment]
Test Outerloop RHEL7.2 Debug please
Test Outerloop Ubuntu14.04 Debug please

**Comment by stephentoub** [comment]
LGTM

**Comment by kapilash** [comment]
Fixed the typo in the commit message.. Innerloop OSX failures are due to 

>  Build machine went offline during the build.

Will launch again

**Comment by kapilash** [comment]
@dotnet-bot 
test innerloop OSX DEBUG please
@dotnet-bot test innerloop OSX RELEASE please

**Comment by kapilash** [comment]
Test Innerloop OSX Debug please

**Comment by kapilash** [comment]
Test Outerloop RHEL7.2 Debug please
Test Outerloop Ubuntu14.04 Debug please

**Comment by stephentoub** [comment]
> Fixed the typo in the commit message

I think you meant "Pending PR comments" not "Pending PR commits".  But it's fine.  Not worth restarting CI for.

**Comment by kapilash** [comment]
> I think you meant "Pending PR comments" not "Pending PR commits".

oops. There was another typo -  it was "prending" which I fixed. 
Thanks for review

**Pull request merged**